### PR TITLE
Signature fix 2

### PR DESCRIPTION
**Pull request by sigmike**
Proposal to fix the current fork between 32 and 64 bits systems.

### CODE REVIEW

**Comment by sandakersmann** [comment]
@rat4 What do you think about this?

**Comment by sandakersmann** [comment]
I have compiled with this patch and everything seems to run as normal on Linux Mint 64-bit.

**Comment by sandakersmann** [comment]
Would be great if someone on Windows 32-bit or Linux 32-bit could compile and test.

File: src/key.cpp [hunk 1]
```diff
@@ -350,28 +350,90 @@ bool CKey::SetCompactSignature(uint256 hash, const std::vector<unsigned char>& v
...
+    unsigned char nLengthBytes = nLengthRet & 0x7f;
+
+    nLengthRet = 0;
+    for (unsigned char i = 0; i < nLengthBytes; i++)
+    {
+        it++;
+        if (it == end)
+            return false;
+        nLengthRet = (nLengthRet << 8) | *it;
+        if (nLengthRet > 0x7f)
```
Comment (sunnyking) [inline src/key.cpp]: scriptSig currently allows up to 500 bytes (c.f. CTransaction::IsStandard()), so here the logic probably should be more flexible, avoid introducing hidden constraint on scriptSig. Something like 
if (nLengthRet > 0x7fffff)
should work

**Comment by sunnyking** [comment]
Looks great! Thanks Mike for the quick fix :)

**Comment by sandakersmann** [comment]
@glv2 Can you push this to peerunity master as well? Would also be great if you could make another release :)

**Pull request merged**